gensim/topic_coherence/probability_estimation.py

```diff
@@ -70,7 +70,7 @@ def unique_ids_from_segments(segmented_topics):
     for s_i in segmented_topics:
         for word_id in itertools.chain.from_iterable(s_i):
             if hasattr(word_id, '__iter__'):
```

Comment:
Are the ids guaranteed to be integers?

Because if they're strings, the `__iter__` could get nasty (you can iterate over strings, yielding individual characters).